client/src/reducers/HomeReducer.ts

```diff
@@ -29,33 +30,49 @@ const initialState: HomeReducerProps = {
   term: "1",
   selectedSchedule: [],
   selectedSections: [],
+  latestState: 0,
 };
 
 export const HomeReducer = (
   state: HomeReducerProps = initialState,
   action: HomeActions
 ): HomeReducerProps => {
+  var latestState;
   switch (action.type) {
     case SWITCHCOMPONENT: {
+      // console.log(action.index, state.latestState, (action.index === 1));
+      if ((action.index > state.latestState + 1 && action.index <= 5) || (state.latestState === 1 && action.index === 2)) {
+        return state;
+      }
       return { ...state, componentIndex: action.index };
     }
     case ADDCOURSESECTIONS: {
-      return { ...state, sections: action.sections };
+      latestState = 1 <= state.latestState ? state.latestState : 2;
+      // console.log("Add course! ", latestState);
```

Comment:
I'm not a huge fan of adding all this logic into the reducer. I think we can validate a lot of this in NextPrevSwitcher component...

Reply:
I second this!